service/db.hpp

```diff
@@ -80,370 +93,309 @@ namespace raven
-                throw_misuse_if_count_return_zero_for_this_statement(select_count_key_statement,
-                                                                     config_load_data.config_key.value().value());
-                execute_statement(select_config_from_key_statement,
-                                  config_load_data.config_key.value().value()) >> functor_receive_data;
-            } else if (config_load_data.config_read_only_key) {
-                throw_misuse_if_count_return_zero_for_this_statement(select_count_readonly_key_statement,
-                                                                     config_load_data.config_read_only_key.value().value());
-                execute_statement(select_config_from_readonly_key_statement,
-                                  config_load_data.config_read_only_key.value().value()) >> functor_receive_data;
-            }
+            throw_misuse_if_count_return_zero_for_this_statement(select_count_config_from_id_statement, config_id.value());
+            execute_statement(select_config_name_statement, config_id.value()) >> functor_receive_data;
+            state = db_state::good;
+        }
+        catch (const sqlite::errors::empty &error) {
+            DLOG_F(ERROR, "misuse of api or wrong key: %s, from sql: %s", error.what(), error.get_sql().c_str());
+            state = db_state::sql_error;
         }
         catch (const sqlite::errors::misuse &error) {
             DLOG_F(ERROR, "misuse of api or wrong key: %s, from sql: %s", error.what(), error.get_sql().c_str());
-            db_answer.request_state = convert_request_state.at(request_state::unknown_key);
+            state = db_state::unknow_config_id;
         }
         catch (const sqlite::sqlite_exception &error) {
             DLOG_F(ERROR, "error: %s, from sql: %s", error.what(), error.get_sql().c_str());
-            db_answer.request_state = convert_request_state.at(request_state::db_error);
+            state = db_state::sql_error;
         }
         catch (const std::exception &error) {
             DLOG_F(ERROR, "%s", error.what());
-            db_answer.request_state = convert_request_state.at(request_state::db_error);
+            state = db_state::fatal_error;
         }
-        return db_answer;
+        return std::move(config_name);
     }
 
-    request_state settings_update(const setting_update &setting_update_data, raven::config_id_st db_id)
+    config_id_st get_config_id(const config_key_st &config_key) noexcept
     {
+        /*
+         * Get the corresponding config_id to the key passed as parameter
+         *
+         * In case an error occur, the state will be set accordingly (`unknow_config_key`, `sql_error`, `fatal_error`)
+         *
+         */
+
         LOG_SCOPE_F(INFO, __PRETTY_FUNCTION__);
+        config_id_st config_id;
+        auto functor_receive_data = [&config_id](int id) { config_id = config_id_st{static_cast<std::size_t>(id)}; };
```

Comment:
Lambda could be noexcept. Not sure we care through (I don't know the sql library enough).

Reply:
not sure either but can't hurt

Reply:
well aparently it hurts since making this lambda noexcept make sql_lite raise an error during the compilation

Reply:
lol wtf
let's keep as is then.